samples/s3-cp/s3-cp.c

```diff
@@ -0,0 +1,403 @@
+                (char *)source_uri_cursor.ptr,
+                aws_error_debug_str(aws_last_error()));
+            s_usage(1);
+        };
+    } else {
+        fprintf(stderr, "An URI for the source object must be supplied.\n");
+        s_usage(1);
+    }
+
+    if (aws_cli_optind < argc) {
+        struct aws_byte_cursor destination_uri_cursor = aws_byte_cursor_from_c_str(argv[aws_cli_optind++]);
+
+        if (aws_uri_init_parse(&ctx->destination_uri, ctx->allocator, &destination_uri_cursor)) {
+            fprintf(
+                stderr,
+                "Failed to parse uri %s with error %s\n",
+                (char *)destination_uri_cursor.ptr,
+                aws_error_debug_str(aws_last_error()));
+            s_usage(1);
+        };
+    } else {
+        fprintf(stderr, "An URI for the destination object must be supplied.\n");
+        s_usage(1);
+    }
+}
+
+/**
+ * Predicate used to decide if the application is ready to exit.
+ * The corresponding condition variable is set when the last
+ * page of ListObjects is received.
+ */
+static bool s_app_completion_predicate(void *arg) {
+    struct app_ctx *app_ctx = arg;
+    return app_ctx->execution_completed;
+}
+
+void setup_logger(struct app_ctx *app_ctx) {
+    struct aws_logger_standard_options logger_options = {
+        .level = app_ctx->log_level,
+        .file = stderr,
+    };
+
+    aws_logger_init_standard(&app_ctx->logger, app_ctx->allocator, &logger_options);
+    aws_logger_set(&app_ctx->logger);
+}
+
+static void s_meta_request_finish(
+    struct aws_s3_meta_request *meta_request,
+    const struct aws_s3_meta_request_result *meta_request_result,
+    void *user_data) {
+
+    (void)meta_request;
+
+    if (meta_request_result->error_code != AWS_ERROR_SUCCESS) {
+        fprintf(
+            stderr,
+            "Meta request failed with error '%s', response status code: %d\n",
+            aws_error_debug_str(meta_request_result->error_code),
+            meta_request_result->response_status);
+    }
+
+    /* copy finished. triggers the condition variable to exit the application. */
+    struct app_ctx *app_ctx = user_data;
+    aws_mutex_lock(&app_ctx->mutex);
+    app_ctx->execution_completed = true;
+    aws_mutex_unlock(&app_ctx->mutex);
+    aws_condition_variable_notify_one(&app_ctx->c_var);
+}
+
+static int s_meta_request_headers(
+    struct aws_s3_meta_request *meta_request,
+    const struct aws_http_headers *headers,
+    int response_status,
+    void *user_data) {
+
+    (void)meta_request;
+    (void)headers;
+    (void)user_data;
+    (void)response_status;
+
+    return 0;
+}
+
+static void s_meta_request_shutdown(void *user_data) {
+
+    (void)user_data;
+}
+
+static int s_meta_request_receive_body(
+    struct aws_s3_meta_request *meta_request,
+    const struct aws_byte_cursor *body,
+    uint64_t range_start,
+    void *user_data) {
+
+    (void)meta_request;
+    (void)body;
+    (void)range_start;
+    (void)user_data;
+
+    return 0;
```

Comment:
if the semantics of this callback are (use all of it or none of it) and return success or error, return AWS_OP_SUCCESS directly so that's clear. If it's the opposite, this might be a bug on the tcp back pressure stuff and should be checked.

Reply:
Removed those callback handlers from sample. They were being used only for debugging and are not needed.